### PR TITLE
Sn save xml

### PR DESCRIPTION
**Pull request by Kert**

### CODE REVIEW

File: src/xsd/map_xml.xsd [hunk 1]
```diff
@@ -106,20 +106,402 @@
...
+      </xs:element>
+	  <xs:element name="field_44D" type="xs:int"/>
+	  <xs:element name="field_451" type="xs:int"/>
+	  <xs:element name="playerHandicap" minOccurs="6" maxOccurs="unbounded">
+        <xs:complexType>
+          <xs:attribute name="id" type="xs:int" use="required"/>
+          <xs:attribute name="handicap" type="xs:int"/>
+        </xs:complexType>
+      </xs:element>
+	  <xs:element name="field_459" minOccurs="6" maxOccurs="unbounded">
```
Comment (jkoppel) [inline src/xsd/map_xml.xsd]: Watch the indentation.

File: src/cpp/shared/game/save.cpp [hunk 2]
```diff
@@ -42,6 +42,18 @@ extern void __fastcall GenerateStandardFileName(char*, char*);
...
 
+template<typename T>
+void ReadArrayFromXML(T &dest, xsd::cxx::tree::sequence<ironfist_save::arrayInt_t> &src) {
+  for (int i = 0; i < src.size(); i++)
+    dest[i] = src.at(i).value();
+}
+
+template<typename T>
+void WriteArrayToXML(xsd::cxx::tree::sequence<ironfist_save::arrayInt_t> &dest, T &src) {
+  for (int i = 0; i < ELEMENTS_IN(src); i++)
```
Comment (jkoppel) [inline src/cpp/shared/game/save.cpp]: ....I have trouble believing that this ELEMENTS_IN call works.

Comment (Kert) [inline src/cpp/shared/game/save.cpp]: Well you can save the game and see that it works in the new savefile though :)

Comment (jkoppel) [inline src/cpp/shared/game/save.cpp]: The behavior I expect to see is that ELEMENTS_IN(src) will always evaluate to 1.

Can you explain why this won't be the case?

Even if it works, you still have to explain it.

Comment (jkoppel) [inline src/cpp/shared/game/save.cpp]: #include<stdio.h>

#define ELEMENTS_IN(x) ((sizeof(x))/(sizeof((x)[0])))


int foo[42];

template<typename T>
void doThing(T &src) {
  printf("Size is %d\n", ELEMENTS_IN(src));
}


int main() {
  doThing(foo);
}


-----------------

Output: 42

Mind blown. I am so confused.

Comment (jkoppel) [inline src/cpp/shared/game/save.cpp]: In my example, it works because doThing got specialized to int[42]

**Pull request merged**